### PR TITLE
Remove dependency from //sky/services/platform to //sky/services/activity

### PR DESCRIPTION
**Pull request by abarth**
Instead of using a static Activity, we can use the one we get when we construct
the service.

### CODE REVIEW

**Comment by abarth** [comment]
@jason-simmons @chinmaygarde

**Comment by abarth** [comment]
The static activity was wrong if there were multiple activities in the same process.

File: sky/shell/platform/android/io/flutter/view/FlutterMain.java [hunk 1]
```diff
@@ -176,7 +176,7 @@ public void connectToService(Context context, Core core, MessagePipeHandle pipe)
         registry.register(HapticFeedback.MANAGER.getName(), new ServiceFactory() {
             @Override
             public void connectToService(Context context, Core core, MessagePipeHandle pipe) {
-                HapticFeedback.MANAGER.bind(new HapticFeedbackImpl(), pipe);
+                HapticFeedback.MANAGER.bind(new HapticFeedbackImpl((android.app.Activity) context), pipe);
```
Comment (jason-simmons) [inline sky/shell/platform/android/io/flutter/view/FlutterMain.java]: What is the lifetime of the service registry?  Is it global to the application, or is there one registry per activity?

Comment (abarth) [inline sky/shell/platform/android/io/flutter/view/FlutterMain.java]: The registry is static.  However, the service provider (which references the registry) is per-FlutterView.  It holds the context and passes it along to this callback.

**Comment by jason-simmons** [comment]
LGTM

**Pull request merged**